### DISCUSSION TITLE
how to get zwave device command class information

### DISCUSSION THREAD

**Discussion by priyanka852** [zwave-js/zwave-js-ui#2827] [Q&A]
In order to control the device with Mqtt like zwave/_CLIENTS/ZWAVE_GATEWAY-test/broadcast/38/0/targetValue as here 38 is a command class of a device, so how can we know the command class of device. is there any api to get the command classes of the device.
Can you please let me know, how to get the command class information of the device, when the device is added.

**Comment by robertsLando** [comment]
Give a look at driverFunction, with that you can call custom async functions and access all the informations you want. Driver function like other apis can be called using MQTT, you can test it from UI

**Reply by robertsLando** [reply]
@priyanka852 specifically you could check `deviceClass`: https://zwave-js.github.io/node-zwave-js/#/api/node?id=deviceclass

it provides both mandatorySupportedCCs and mandatoryControlledCCs
